src/chain_actions.cc

```diff
@@ -122,9 +122,31 @@ namespace dd
                 cxmax = int(xcenter + fixed_width / 2.0);
                 cymin = int(ycenter - fixed_height / 2.0);
                 cymax = int(ycenter + fixed_height / 2.0);
+
+                if (cxmin < 0)
+                  {
+                    cxmax += -cxmin;
+                    cxmin = 0;
+                  }
+                if (cymin < 0)
+                  {
+                    cymax += -cymin;
+                    cymin = 0;
+                  }
+                if (cxmax > img.cols)
+                  {
+                    cxmin -= cxmax - img.cols;
```

Comment:
Here you may run into cxmin < 0 again, and this would mean the crop cannot be set to the proper size. I believe this requires an additional and final 'check' that would fail the crop ?

Reply:
Or may this can only happen if the original image has size equal or lower than the requested crop ?

Reply:
Yes, it can only happen if image width < requested fixed_width